default_middlewares.go

```diff
@@ -0,0 +1,47 @@
+package fuego
+
+import (
+	"log/slog"
+	"net/http"
+)
+
+type LoggingConfig struct {
+	DisableRequest  bool
+	DisableResponse bool
+
+	RequestLogger  func(w http.ResponseWriter, r *http.Request)
+	ResponseLogger func(w http.ResponseWriter, r *http.Request)
+}
+
+func (l *LoggingConfig) Disabled() bool {
+	return l.DisableRequest && l.DisableResponse
+}
+
+var defaultLoggingConfig = LoggingConfig{
+	RequestLogger:  logRequest,
+	ResponseLogger: logResponse,
+}
+
+func defaultLoggingMiddleware(s *Server) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if !s.loggingConfig.DisableRequest {
+				s.loggingConfig.RequestLogger(w, r)
+			}
+
+			next.ServeHTTP(w, r)
+
+			if !s.loggingConfig.DisableResponse {
+				s.loggingConfig.ResponseLogger(w, r)
+			}
+		})
+	}
+}
+
+func logRequest(w http.ResponseWriter, r *http.Request) {
+	slog.Info("<- request")
+}
+
+func logResponse(w http.ResponseWriter, r *http.Request) {
+	slog.Info("response ->")
+}
```

Comment:
More complicated but I'd like to see

- Method
- Path
- Timestamp

For another PR:
- Corresponding request ID
- Returned status code
- Timer with the request duration

Reply:
You’ll probably need to wrap request or response as well to get the duration as well

Reply:
This is also where I stopped my original implementation. I’m not really sure it’s worth it to allow users to set the functions themselves. By that point they’re being forced to implement a specific function signature where it’s probably easier to just disable the middleware and bring their own.

Reply:
For duration you can just have a timer in the middleware. But I'm not sure that current implementation allows to share data between the first and the second function

Reply:
The usual way is indeed a time.Now before the call and a time.Since after the call.

The start time doesn't need to be in the context